wzdx/standard_to_enhanced/planned_events_translator.py

```diff
@@ -0,0 +1,393 @@
+    # I included a skeleton of the message, fill out all required fields and as many optional fields as you can. Below is a link to the spec page for a road event
+    # https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/spec-content/objects/RoadEvent.md
+
+    core_details = {}
+
+    # data_source_id
+    # Leave this empty, it will be populated by add_ids_v3
+    core_details['data_source_id'] = ''
+
+    # Event Type ['work-zone', 'detour']
+    core_details['event_type'] = event.get('type')
+
+    # road_name
+    road_names = [detail.get('road_name')]
+    core_details['road_names'] = road_names
+
+    # direction
+    direction = None
+    for road_name in road_names:
+        direction = wzdx_translator.parse_direction_from_street_name(road_name)
+        if direction:
+            break
+    if not direction:
+        direction = polygon_tools.get_road_direction_from_coordinates(
+            geometry.get('coordinates'))
+    if not direction:
+        return None
+    core_details['direction'] = direction
+
+    # Relationship
+    core_details['relationship'] = {}
+
+    # description
+    core_details['description'] = header.get('description')
+
+    # creation_date
+    core_details['creation_date'] = date_tools.get_iso_string_from_unix(
+        source.get('creation_timestamp'))
+
+    # update_date
+    core_details['update_date'] = date_tools.get_iso_string_from_unix(
+        source.get('last_updated_timestamp'))
+
+    properties['core_details'] = core_details
+
+    properties['lanes'] = additional_info.get('lanes', [])
+
+    # restrictions
+    properties['restrictions'] = additional_info.get('restrictions', [])
+
+    filtered_properties = copy.deepcopy(properties)
+
+    for key, value in properties.items():
+        if not value and key not in ['road_event_id', 'data_source_id']:
+            del filtered_properties[key]
+
+    feature = {}
+    feature['type'] = "Feature"
+    feature['properties'] = filtered_properties
+    feature['geometry'] = geometry
+    feature['id'] = event.get('source', {}).get('id', uuid.uuid4())
+    print(feature)
+
+    return feature
+
+
+# Parse Icone Incident to WZDx
+def parse_work_zone(incident):
+    if not incident or type(incident) != dict:
+        return None
+
+    event = incident.get('event')
+
+    source = event.get('source')
+    header = event.get('header')
+    detail = event.get('detail')
+    additional_info = event.get('additional_info', {})
+
+    geometry = {}
+    geometry['type'] = "LineString"
+    geometry['coordinates'] = event.get('geometry')
+    properties = {}
+
+    # I included a skeleton of the message, fill out all required fields and as many optional fields as you can. Below is a link to the spec page for a road event
+    # https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/spec-content/objects/RoadEvent.md
+
+    core_details = {}
+
+    # # id
+    # # Leave this empty, it will be populated by add_ids_v3
+    # core_details['road_event_id'] = ''
```

Comment:
Should this be setting the `road_event_id` field to empty, or leaving the whole field out?